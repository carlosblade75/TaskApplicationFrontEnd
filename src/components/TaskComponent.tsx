import React, { useContext, useState } from 'react';
import { TaskContext } from '../Context/TaskContext'

type PropsTask = {
    id: number,
    description: string,
    isCompleted: boolean
}

export const TaskComponent = (props: PropsTask) => {

    const [value, setValue] = useState(props.isCompleted);

    const {id, description} = props;

    const { checkUpdate } = useContext(TaskContext);

    const handleToggleChange= () => {
        console.log('Cmabiando')
        checkUpdate({ id, description, isCompleted: !value});
        setValue(!value)
    }

    return (
        <div>
            <input type="hidden" value={id}/>
            <input type="checkbox" checked={value} onChange= { handleToggleChange }/>
            <label>{description}</label>
        </div>
    )
}