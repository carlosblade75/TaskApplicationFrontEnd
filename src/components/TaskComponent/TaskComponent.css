.CheckBox {
    font-size: 16px;
    margin-right: 15px;
    cursor: pointer;
}

.CheckBoxLabel {
    padding-left: 10px;
    font-size: 18px;
    font-family: Arial, Helvetica, sans-serif;
}

.isCompleted {
    text-decoration: line-through;
    color: rgb(35, 206, 72);
}
