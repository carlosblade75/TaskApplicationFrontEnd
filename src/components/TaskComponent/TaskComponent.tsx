import React, { useContext, useState } from 'react';
import { TaskContext } from '../../Context/TaskContext'
import './TaskComponent.css'
import PropTypes from "prop-types";

type PropsTask = {
    id: number,
    description: string,
    isCompleted: boolean
}

export const TaskComponent = (props: PropsTask) => {

    const [value, setValue] = useState(props.isCompleted);

    const {id, description} = props;

    const { checkUpdate } = useContext(TaskContext);

    const handleToggleChange= () => {
        checkUpdate({ id, description, isCompleted: !value});
        setValue(!value)
    }

    return (
        <div className="form-check ContainerTask">
            
            <input  type="hidden" 
                    value={id}
            />
            
            <label  className={"form-check-label CheckBoxLabel " + (value ? "isCompleted" : "")}
                    onClick={ handleToggleChange }>
                {description}
            </label>
            
        </div>
    )
}

TaskComponent.propTypes = {
    description : PropTypes.string.isRequired,
    id: PropTypes.number.isRequired
}

TaskComponent.defaultProps = {
    isCompleted: false
}