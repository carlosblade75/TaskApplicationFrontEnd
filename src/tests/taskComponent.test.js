import React from 'react'
import '@testing-library/jest-dom'
import { TaskComponent } from "../components/TaskComponent/TaskComponent"
import { shallow } from 'enzyme'

describe('Testing for <TaskComponent/>', () => {

    const id = 1
    const description = 'Task Description'
    const wrapper = shallow( <TaskComponent description = {description} id = { id } isCompleted = {false} /> )

    test('should show <TaskComponent/> properly', () => {
        expect(  wrapper ).toMatchSnapshot()
    });

    test('should show the description properly', () => {

        const wrapper = shallow( <TaskComponent description = {description} id = { id }  /> )

        const descriptionInComponent = wrapper.find('label').text()

        expect( descriptionInComponent ).toBe( description )

    });

    test('should show the task as pending', () => {
        
        const className = wrapper.find('label').prop('className')

        expect( className.includes('form-check-label CheckBoxLabel')).toBe( true )
        expect( className.includes('isCompleted')).toBe( false )
    })

    test('should shows the task as line-through when is completed', () => {
        
        const wrapper = shallow( <TaskComponent description = {description} id = { id } isCompleted={true} /> )
        
        const className =  wrapper.find('label').prop('className')

        expect( className.includes('form-check-label CheckBoxLabel')).toBe( true )
        expect( className.includes('isCompleted')).toBe( true )
    })
    
})