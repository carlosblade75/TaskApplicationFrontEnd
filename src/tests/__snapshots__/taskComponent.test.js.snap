// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing for <TaskComponent/> should show <TaskComponent/> properly 1`] = `
<div
  className="form-check ContainerTask"
>
  <input
    type="hidden"
    value={1}
  />
  <label
    className="form-check-label CheckBoxLabel "
    onClick={[Function]}
  >
    Task Description
  </label>
</div>
`;
